src/tests/Microsoft.PowerFx.Core.Tests/ImmutabilityTests.cs

```diff
@@ -16,6 +16,9 @@ namespace Microsoft.PowerFx.Core.Tests
     /// </summary>
     internal class ImmutabilityTests : PowerFxTest
     {
+        // Include non-public types
+        // Include non-public properties!
+        // Are private-setters ok?
         public static void CheckImmutability(Assembly asm)
```

Comment:
This test passes and asserts, but it is limited. 
The new test checks many more things (which currently fail), so it has fewer asserts.